### PR TITLE
Updates of training utilities

### PR DESCRIPTION
**Pull request by miaecle**
This PR adds:

- Reformat functions for calculating basic cell geometry properties
- Add weight option for components of loss function
- Add summary writer for the main `train` method
- Fix some bugs in the data preprocessing pipeline
- A method to combine multiple datasets

### CODE REVIEW

**Comment by miaecle** [comment]
@smguo I copied some of the training utilities you used in `run_train.py`, also there is a change in adjusting weights for the components of loss function, please take a look and make sure there won't be conflicts.

File: HiddenStateExtractor/vq_vae_supp.py [hunk 1]
```diff
@@ -145,14 +145,16 @@ def prepare_dataset_v2(dat_fs,
     dataset = np.stack([tensors[key] for key in ts_keys], 0)
     return dataset, ts_keys
 
-def reorder_with_trajectories(dataset, relations, seed=None):
+def reorder_with_trajectories(dataset, relations, seed=None, w_a=1.1, w_t=0.1):
```
Comment (mattersoflight) [inline HiddenStateExtractor/vq_vae_supp.py:148]: Hi @miaecle , do you call this function only after first 100 epochs to 'burn-in' the model without matching loss? Is the corresponding training loop in this PR?

Comment (miaecle) [reply HiddenStateExtractor/vq_vae_supp.py:148]: Actually no, this is called before all training.
